src/registered.js

```diff
@@ -58,26 +58,34 @@ module.exports = async function (request, h) {
     codeHosting: [{
       url: url
     }],
+    alternativeId: ipa,
   };
 
   const apiURL = config.apiURL.replace(/\/$/, '');
 
   try {
-    const res = await fetch(`${apiURL}/publishers`, {
-      method: 'POST',
-      body: JSON.stringify(apiPayload),
+    const getPublisherResp = await fetch(`${apiURL}/publishers/${ipa}`, {
+      method: 'GET',
       headers: {'Content-Type': 'application/json', 'Authorization': `Bearer ${apiPasetoKey}`},
     });
 
-    const data = await res.json();
+    const publisher = await getPublisherResp.json();
 
-    if (!res.ok) {
-      throw data.validationErrors?.map(error => `
-        Valore non valido: <strong>${error.value}</strong> per il campo: <strong>${error.field}</strong>
-        con la regola: <strong>${error.rule}</strong><br>
-      `).join('') || data.detail;
+    switch (getPublisherResp.status) {
+    case 200:
+      if (publisher.alternativeId !== ipa) {
+        throw new Error('Si è verificato un errore imprevisto durante la verifica dell\'esistenza del publisher');
+      }
+
+      await updateExistingPublisher(apiURL, apiPayload, apiPasetoKey, publisher.id);
+      break;
+
+    case 404:
+      await createPublisher(apiURL, apiPayload, apiPasetoKey);
+      break;
     }
```

Comment:
This will fail silently when `GET /publishers/id` returns !404 || !200, leaving the API and the local database unsynced.

Reply:
Ok, but i don't know if we should maintain local DB how it behaves, due to the fact that we will deprecate it.
Otherwise, we've also to do update in the local db.
Right now, I will update the code to reflect writing only in case of 404.

Reply:
I'm not sure I get what you mean, but we need to update the local database every time we make a change via the API, that is when PATCH or POST are successful.

Unless I'm mistaken, with the last change `updateExistingPublisher()` would add a `codeHosting` via the API, but it won't be added to the local database.

Reply:
Are we sure to also update the local db part?
It will be deprecated, IMHO we should maintain the old way of how it works, no?
Do you prefer also to update the db behavior and add the update part? @bfabio

Reply:
Yep, that's fine.